001-IntroToKubernetes/Student/08-storage.md

```diff
@@ -17,14 +17,15 @@ In this challenge we will be creating Azure data disks and using the Kubernetes
 	- Delete the MongoDB pod created in Challenge 6. 
 		- Kubernetes should automatically create a new pod to replace it. 
 		- Connect to the new pod and you should observe that the “contentdb” database is no longer there since the pod was not configured with persistent storage.
-	- Delete the MongoDB deployment
-	- Note that your data disappears!
-- Redeploy MongoDB with dynamic persistent storage
 	- Delete your existing MongoDB deployment
-	- Create a Persistent Volume Claim (PVC).
-	- Examine the Persistent Volume (PV) created.
+- Redeploy MongoDB with dynamic persistent storage
```

Comment:
Some general comments & questions on this challenge...

The note & steps after it talk about having to take Availability Zones into account when provisioning storage on an AKS cluster.  This adds a few more steps of complexity to this challenge that could be considered beyond the scope of the challenge itself.  For example, the concept of forcing a pod to be scheduled on a specific node, and thus the need to create the "storage class" to make this possible.

If the user didn't create their AKS cluster using Avail Zones, then a lot of these steps are unneeded.  That being said, we DID say they should use AZs in Challenge 3 when creating AKS, so that puts us on the hook to cover the complexity here in Challenge 8.

I see that you are providing students with a "template-storageclass.yaml".  I think it's a good idea to "give" this to the students as its not core to the challenge.  It's really a "hack" to get this to work in the Azure environment.  I would perhaps expand upon the instruction on line 23 "Create a new storage Class..." and explain:
1) what a storage class is
2) why you need it here
3) and "oh, we're providing this file to you in the Resources folder"

On line 28, I'd change "pods" to "pod" since there should only be a single replica in the deployment.

Reply:
Updated. (As it turns out, we don't even need to lock into a single region.)